src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs

```diff
@@ -41,10 +31,10 @@ public static TypeSymbolWithAnnotations InferBestType(ImmutableArray<BoundExpres
             // SPEC:    If no such S exists, the expressions have no best common type.
 
             // All non-null types are candidates for best type inference.
-            var candidateTypes = new HashSet<TypeSymbolWithAnnotations>(TypeSymbolWithAnnotations.EqualsComparer.Instance);
+            var candidateTypes = new HashSet<TypeSymbol>(conversions.IncludeNullability ? TypeSymbol.EqualsIncludingNullableComparer : TypeSymbol.EqualsConsiderEverything);
```

Comment:
>TypeSymbol [](start = 45, length = 10)

How do we deal with top level nullability?

Reply:
This overload is only called by initial binding where nullability is ignored. I've added an assert here and remove the condition.

---
In reply to: [168043939](https://github.com/dotnet/roslyn/pull/23981#discussion_r168043939) [](ancestors = 168043939)